src/user/user.repository.ts

```diff
@@ -0,0 +1,49 @@
+import { Injectable } from '@nestjs/common';
+import { PrismaService } from '../prisma/prisma.service';
+import { Prisma, User } from '@prisma/client';
+import { UserDto } from './dto/user-response';
+
+@Injectable()
+export class UserRepository {
+  constructor(private prisma: PrismaService) {}
+
+  async getUser(where: Prisma.UserWhereUniqueInput): Promise<UserDto> {
+    const user = await this.prisma.user.findUnique({
+      where,
+    });
+
+    return UserDto.from(user);
+  }
+
+  getUserWithPassword(where: Prisma.UserWhereUniqueInput): Promise<User> {
+    return this.prisma.user.findUnique({
+      where,
+    });
+  }
+
+  async createUser(data: Prisma.UserCreateInput): Promise<UserDto> {
```

Comment:
Here you are returning `Promise<User>` and here `Promise<UserDto>` - I think I'd just return prisma `User` type here. It comes with some disadvantages though but I think it will be better right now.

Use `UserDto` (`UserResponse` after applying suggestion above) in controller when returning data.

Reply:
Before changing the implementation check out [this comment](https://github.com/KubaBaniak/RecipeAppNestjs/pull/25/files#r1265328378).